cpp/src/interop/from_arrow_host.cu

```diff
@@ -0,0 +1,445 @@
+  // mask-to-bools expects the mask to be bitmask_type aligned/padded
+  auto data = rmm::device_buffer(buffer_length, stream, mr);
+  CUDF_CUDA_TRY(cudaMemcpyAsync(data.data(),
+                                reinterpret_cast<uint8_t const*>(data_buffer),
+                                buffer_length,
+                                cudaMemcpyDefault,
+                                stream.value()));
+  auto out_col = mask_to_bools(static_cast<bitmask_type*>(data.data()),
+                               input->offset,
+                               input->offset + input->length,
+                               stream,
+                               mr);
+
+  auto const has_nulls = skip_mask ? false : input->buffers[validity_buffer_idx] != nullptr;
+  if (has_nulls) {
+    auto out_mask = detail::copy_bitmask(static_cast<bitmask_type*>(get_mask_buffer(input)->data()),
+                                         input->offset,
+                                         input->offset + input->length,
+                                         stream,
+                                         mr);
+
+    out_col->set_null_mask(std::move(out_mask), input->null_count);
+  }
+
+  return out_col;
+}
+
+template <>
+std::unique_ptr<column> dispatch_copy_from_arrow_host::operator()<cudf::string_view>(
+  ArrowSchemaView* schema, ArrowArray const* input, data_type type, bool skip_mask)
+{
+  if (input->length == 0) { return make_empty_column(type_id::STRING); }
+
+  const void* offset_buffers[2] = {nullptr, input->buffers[fixed_width_data_buffer_idx]};
+  ArrowArray offsets_array      = {
+         .length     = input->offset + input->length + 1,
+         .null_count = 0,
+         .offset     = 0,
+         .n_buffers  = 2,
+         .n_children = 0,
+         .buffers    = offset_buffers,
+  };
+
+  size_type const char_data_length =
+    reinterpret_cast<int32_t const*>(offset_buffers[1])[input->length + input->offset];
+  const void* char_buffers[2] = {nullptr, input->buffers[2]};
+  ArrowArray char_array       = {
+          .length     = char_data_length,
+          .null_count = 0,
+          .offset     = 0,
+          .n_buffers  = 2,
+          .n_children = 0,
+          .buffers    = char_buffers,
+  };
+
+  nanoarrow::UniqueSchema offset_schema;
+  NANOARROW_THROW_NOT_OK(ArrowSchemaInitFromType(offset_schema.get(), NANOARROW_TYPE_INT32));
+
+  nanoarrow::UniqueSchema char_data_schema;
+  NANOARROW_THROW_NOT_OK(ArrowSchemaInitFromType(char_data_schema.get(), NANOARROW_TYPE_INT8));
+
+  ArrowSchemaView view;
+  NANOARROW_THROW_NOT_OK(ArrowSchemaViewInit(&view, offset_schema.get(), nullptr));
+  auto offsets_column =
+    this->operator()<int32_t>(&view, &offsets_array, data_type(type_id::INT32), true);
+  NANOARROW_THROW_NOT_OK(ArrowSchemaViewInit(&view, char_data_schema.get(), nullptr));
+  auto chars_column = this->operator()<int8_t>(&view, &char_array, data_type(type_id::INT8), true);
```

Comment:
IIUC the reason that the first elements of the length-2 buffer arrays inserted as the buffers to `offsets_buffers` and `char_buffers` above are set to nullptr is because those are the mask and we're relying on the mask to get set by these calls to dispatch to the int32/int8 overloads of this operator. Is that right? If so, could we add a comment explaining how this overload is leveraging overloads for other data types? I had to read this twice before I realized what was going on.

Reply:
added some comments. let me know if I should add more